packages/api/src/user.js

```diff
@@ -87,8 +151,10 @@ async function loginOrRegister (request, env) {
   if (env.MODE === NO_READ_OR_WRITE) {
     return maintenanceHandler()
   } else if (env.MODE === READ_WRITE) {
-    // @ts-ignore
     user = await env.db.upsertUser(parsed)
+    if (user.inserted) {
+      await initializeNewUser(env, { ...user, id: user.id.toString() })
```

Comment:
I think you can write a PGSQL function that uses transactions and call it via postgrest. I haven't tried it personally tho.

Reply:
@yusefnapora In this case, I don't think you can? (could be wrong). In this case, I'd want to BEGIN a transaction and do the upsert, then (in js with a network, not in sql) invoke some stripe apis, and only if the stripe api call succeeds would i COMMIT the transaction. I don't think you could do that in SQL fn alone, unless you can have two functions-- one to BEGIN the transaction, and then the second function async later calls COMMIT

Reply:
ah yeah, that makes sense. I think it's generally a Bad Thing to hold a transaction open while doing network I/O anyway, so I think the `inserted` flag is a better fit for this anyway

Reply:
yeah its a Bad Thing if there's a better way. but idk it also seems like the least bad thing compared to having partial application of the initialization and no way to rollback. It would be a problem under high writes probably, but IMO this part should be fairly low traffic on write.

fwiw i think async transactions isn't always bad thing e.g. https://en.wikipedia.org/wiki/Two-phase_commit_protocol